### ISSUE TITLE
[flang][multi image] Unexpected error 'arith.cmpi' op requires all operands to have the same type

### ISSUE THREAD

**Issue by sscalpone** [llvm/llvm-project#207302]
Perhaps related to #192944.

`% cat co.f90`

```fortran
! Same f18 error: 'arith.cmpi' op requires all operands to have the same type.
Real a[*]
Print *, This_Image(a)/=[1]
End
```

```
% flang -c -fcoarray co.f90
warning: Support for multi image Fortran features is still experimental and in development.
error: loc("co.f90":3:10): 'arith.cmpi' op requires all operands to have the same type
error: verification of lowering to FIR failed
```

**Comment by JDPailleux** [comment]
Yes probably ! Thanks @sscalpone

**Comment by bonachea** [comment]
Likely fix: #208429
